test/unit/puppetlabs/services/request_handler/request_handler_core_test.clj

```diff
@@ -292,45 +290,41 @@
-                      custom-vcs
-                      tk-scheduler/scheduler-service]]
-        (jruby-bootstrap/with-puppetserver-running-with-services
-         app services {:jruby-puppet {:max-active-instances 1}}
-         (jruby-testutils/wait-for-jrubies app)
-         (let [in-catalog-request-future (promise)
-               jruby-service (tk-app/get-service app :JRubyPuppetService)]
-           (future
-            (deliver in-catalog-request-future true)
-            (testutils/get-catalog))
-           (deref in-catalog-request-future)
-           (is (deref first-promise 10000 false))
-           ; Because we are blocking inside current-code-id, which happens to be
-           ; used during a jruby request, we can assert that there will be no
-           ; jruby instances left in the pool.
-           (is (zero? (jruby-protocol/free-instance-count jruby-service)))
-           (deliver second-promise true)))))))
+     (let [first-promise (promise)
+           second-promise (promise)
+           custom-vcs (service vc/VersionedCodeService
+                               []
+                               (current-code-id [_ _]
+                                                (deliver first-promise true)
+                                                (deref second-promise 5000 false))
+                               (get-code-content [_ _ _ _]
+                                                 nil))
+           services [master-service/master-service
+                     jruby-service/jruby-puppet-pooled-service
+                     profiler/puppet-profiler-service
+                     handler-service/request-handler-service
+                     ps-config/puppet-server-config-service
+                     jetty9/jetty9-service
+                     ca-service/certificate-authority-service
+                     authorization-service/authorization-service
+                     routing-service/webrouting-service
+                     custom-vcs
+                     tk-scheduler/scheduler-service]]
+       (jruby-bootstrap/with-puppetserver-running-with-services-and-mock-jrubies
+        app services {:jruby-puppet {:max-active-instances 1}}
+        (jruby-testutils/wait-for-jrubies app)
+        (let [in-catalog-request-future (promise)
+              jruby-service (tk-app/get-service app :JRubyPuppetService)]
+          (future
+           (deliver in-catalog-request-future true)
+           (testutils/get-catalog))
+          (deref in-catalog-request-future)
+          (is (deref first-promise 10000 false))
+          ; Because we are blocking inside current-code-id, which happens to be
+          ; used during a jruby request, we can assert that there will be no
+          ; jruby instances left in the pool.
+          (is (zero? (jruby-protocol/free-instance-count jruby-service)))
+          (deliver second-promise true))))))
 
 (deftest request-handler-test
     (let [dummy-service (reify jruby-protocol/JRubyPuppetService
```

Comment:
Done with this commit, nothing here other than calls to the macro that I will evaluate when I'm adding docstrings.